spec.html

```diff
@@ -7649,58 +7649,42 @@ <h1>Jobs and Job Queues</h1>
-    <p>Each agent has its own set of named Job Queues.  All references to a named job queue in this specification denote the named job queue of the surrounding agent.</p>
-    <emu-table id="table-26" caption="Required Job Queues">
-      <table>
-        <tbody>
-        <tr>
-          <th>
-            Name
-          </th>
-          <th>
-            Purpose
-          </th>
-        </tr>
-        <tr>
-          <td>
-            ScriptJobs
-          </td>
-          <td>
-            Jobs that validate and evaluate ECMAScript |Script| and |Module| source text. See clauses 10 and 15.
-          </td>
-        </tr>
-        <tr>
-          <td>
-            PromiseJobs
-          </td>
-          <td>
-            Jobs that are responses to the settlement of a Promise (see <emu-xref href="#sec-promise-objects"></emu-xref>).
-          </td>
-        </tr>
-        </tbody>
-      </table>
-    </emu-table>
-    <p>A request for the future execution of a Job is made by enqueueing, on a Job Queue, a PendingJob record that includes a Job abstract operation name and any necessary argument values. When there is no running execution context and the execution context stack is empty, the ECMAScript implementation removes the first PendingJob from a Job Queue and uses the information contained in it to create an execution context and starts execution of the associated Job abstract operation.</p>
-    <p>The PendingJob records from a single Job Queue are always initiated in FIFO order. This specification does not define the order in which multiple Job Queues are serviced. An ECMAScript implementation may interweave the FIFO evaluation of the PendingJob records of a Job Queue with the evaluation of the PendingJob records of one or more other Job Queues. An implementation must define what occurs when there are no running execution context and all Job Queues are empty.</p>
-    <emu-note>
-      <p>Typically an ECMAScript implementation will have its Job Queues pre-initialized with at least one PendingJob and one of those Jobs will be the first to be executed. An implementation might choose to free all resources and terminate if the current Job completes and all Job Queues are empty. Alternatively, it might choose to wait for a some implementation specific agent or mechanism to enqueue new PendingJob requests.</p>
-    </emu-note>
-    <p>The following abstract operations are used to create and manage Jobs and Job Queues:</p>
+    <p>Jobs are scheduled for execution by ECMAScript host environments. Such operations accept a PendingJob Record _pending_ as a parameter. These abstract operations schedule the PendingJob to be performed at some future time. Their implementations must conform to the following requirements:</p>
```

Comment:
> Such operations

You removed the reference to operations, so this doesn't make sense any more.

Reply:
OK, I've pushed a version which restores the "such as HostEnqueuePromiseJob" wording, but giving more context about how hosts can also enqueue jobs. My understanding of the agreement reached with @erights in the June 2019 TC39 meeting was that we can do this layering change as long as we don't add these kinds of extra restrictions to hosts, and generally preserve the notion of jobs in general. That's why many things in this PR are as they are (for example, the general EnqueueJob abstract operation, rather than specifically EnqueuePromiseJob). If you want to make things more specific, you might want to follow up with him first.

Reply:
Since "enqueue a job" is just an awkward wrapper around "call an abstract operation", I don't see how we would possibly restrict hosts from doing so.

I don't think preserving the notion of jobs in general is a great idea, given the disparate requirements of different job-like things (tasks, microtasks, compound microtasks, IndexedDB cleanup tasks, whatever finalization group callbacks end up as, etc.). As I stated in https://github.com/tc39/ecma262/pull/1597#discussion_r382071063 I think it'd be great if the editors wanted to simplify this away from the current over-general structure and extra abstractions.

Reply:
@domenic The editors generally like the suggestion and are going to take a stab at it in a follow-up PR.

Reply:
My priority here is to fix the layering with HTML, enabling Promises and WeakRefs to be specified in a readable way. I don't really want to land multiple things in HTML while the JS spec iterates here, so please let me know when the editors arrive at a final conclusion here on the wording so I can pick up the HTML side. I've done the best I can to incorporate the committee's constraints into something we can get consensus on.